planner/core/task.go

```diff
@@ -935,6 +935,54 @@ func (p *PhysicalLimit) sinkIntoIndexLookUp(t task) bool {
 	return true
 }
 
+func (p *PhysicalLimit) sinkIntoIndexMerge(t task) bool {
```

Comment:
There's some type assertion in the codes. It might introduce more code complexity for the pull. We can left it for next pulls. Such as add a new common interface for the two struct.